### PR TITLE
docs(api): Settle in to leave Labware.default_magdeck_engage_height alone, for now

### PR DESCRIPTION
**Pull request by SyntaxColoring**
# Overview

This "closes" RCORE-539 in a kind of cheaty way: by determining that we don't actually need to do it right now. 😅 

# Changelog

* Behaviorally: keep `labware.default_magdeck_engage_height` the same way it has been.
* Add a public docstring to `labware.default_magdeck_engage_height` to explicitly discourage its use.
* Do a light simplification of core internals to take advantage of the above.

Rationale:

* `labware.default_magdeck_engage_height`'s implementation is "correct." Because of a panoply of misbehaviors in the things surrounding it (RSS-109) and its lack of explicit compensation for those misbehaviors, it will return "wrong" results.
* We have neither the desire nor the ability to fix any of that in this release.
* But, from an in-person discussion with @mcous, the good news is that we think this property is only incidentally public. Although it's always been on https://docs.opentrons.com, it hasn't had a docstring before, and we don't expect that any protocols in the field are heavily relying upon it.
* So, by discouraging it, we reduce API surface area that we need to implement correctly for PAPIv2.14, and we buy ourselves some more room in the future to resolve RSS-109 across the board.

# Test Plan

I've smoke-tested with `opentrons_simulate` that in the legacy PAPI backend, `magnetic_module.engage()`:

* still raises an error when no labware is present
* still raises an error when a labware is present but is missing a default engage height
* still succeeds when a labware is present that has a default engage height

# Review requests

* Is the new docstring on `Labware.default_magdeck_engage_height` is sufficiently discouraging that, in the future, we will feel free to modify its behavior however we want, without regard for backwards compatibility?

# Risk assessment

Low because the behavioral changes are tightly contained.

### CODE REVIEW

**Review by mcous — APPROVED** [review]
Love it 🦶🏈

File: api/src/opentrons/protocol_api/labware.py [hunk 1]
```diff
@@ -395,10 +395,28 @@ def quirks(self) -> List[str]:
...
+    # Remove warning once we resolve RSS-109 more broadly.
     @property  # type: ignore
     @requires_version(2, 0)
     def magdeck_engage_height(self) -> Optional[float]:
+        """Return the default magnet engage height that
+        :py:meth:`MagneticModuleContext.engage` will use for this labware.
+
+        .. warning::
+            This currently returns confusing and unpredictable results that do not
+            necessarily match what :py:meth:`MagneticModuleContext.engage` will
```
Comment (ecormany) [inline api/src/opentrons/protocol_api/labware.py]:
```suggestion
        :py:meth:`.MagneticModuleContext.engage` will use for this labware.

        .. warning::
            This currently returns confusing and unpredictable results that do not
            necessarily match what :py:meth:`.MagneticModuleContext.engage` will
```
doesn't link properly without the leading dot

File: api/src/opentrons/protocol_api/labware.py [hunk 2]
```diff
@@ -395,10 +395,28 @@ def quirks(self) -> List[str]:
...
+            actually choose for its default height.
+
+            The confusion is related to how this height's units and origin point are
+            defined, and differences between Magnetic Module generations.
+
+            For now, we recommend you avoid accessing this property directly.
```
Comment (ecormany) [inline api/src/opentrons/protocol_api/labware.py:423]: The docstring as a whole and especially the warning seem clear and candid. If honesty is the best policy, this is the way to do it!

**Review by ecormany — APPROVED** [review]
One tiny syntax fix but this looks good. 🥾 🏈

File: api/src/opentrons/protocol_api/labware.py (hunk 1 above)
Comment (SyntaxColoring) [reply api/src/opentrons/protocol_api/labware.py]: Nice catch. Serves me right for trying to write ReST syntax from memory.

**Comment by SyntaxColoring** [comment]
Investigating why this is failing on the Python 3.10 CI job.

Edit: It was because my refactors broke the tests. I don't have the brain power to figure out how to adjust the tests at the moment, so I'm just undoing the refactors.

**Pull request merged**